File: app/src/main/java/org/mesonet/app/site/mesonetdata/MesonetModel.java
package org.mesonet.app.site.mesonetdata;


import org.mesonet.app.formulas.DefaultUnits;
import org.mesonet.app.formulas.UnitConverter;

public class MesonetModel implements DefaultUnits
{
    Integer STNM;
    Long TIME;
    Number RELH;
    Number TAIR;
    Number WSPD;
    Number WVEC;
    Number WDIR;
    Number WDSD;
    Number WSSD;
    Number WMAX;
    Number RAIN;
    Number PRES;
    Number SRAD;
    Number TA9M;
    Number WS2M;
    Number TS10;
    Number TB10;
    Number TS05;
    Number TS25;
    Number TS60;
    Number TR05;
    Number TR25;
    Number TR60;
    Number RAIN_24H;
    Number RAIN_24;
    String STID;
    Integer YEAR;
    Integer MONTH;
    Integer DAY;
    Integer HOUR;
    Integer MINUTE;
    Integer SECOND;



    @Override
    public UnitConverter.TempUnits GetDefaultTempUnit()
    {
        return UnitConverter.TempUnits.kCelsius;
    }



    @Override
    public UnitConverter.LengthUnits GetDefaultLengthUnit()
    {
        return UnitConverter.LengthUnits.kMm;
    }



    @Override
    public UnitConverter.SpeedUnits GetDefaultSpeedUnit()
    {
        return UnitConverter.SpeedUnits.kMps;
    }



    @Override
    public UnitConverter.PressureUnits GetDefaultPressureUnit()
    {
        return UnitConverter.PressureUnits.kMb;
    }
}

File: app/src/main/java/org/mesonet/app/site/mesonetdata/MesonetModel.kt
package org.mesonet.app.site.mesonetdata

import org.mesonet.app.formulas.DefaultUnits
import org.mesonet.app.formulas.UnitConverter


class MesonetModel : DefaultUnits {
    internal var STNM: Int? = null
    internal var TIME: Long? = null
    internal var RELH: Number? = null
    internal var TAIR: Number? = null
    internal var WSPD: Number? = null
    internal var WVEC: Number? = null
    internal var WDIR: Number? = null
    internal var WDSD: Number? = null
    internal var WSSD: Number? = null
    internal var WMAX: Number? = null
    internal var RAIN: Number? = null
    internal var PRES: Number? = null
    internal var SRAD: Number? = null
    internal var TA9M: Number? = null
    internal var WS2M: Number? = null
    internal var TS10: Number? = null
    internal var TB10: Number? = null
    internal var TS05: Number? = null
    internal var TS25: Number? = null
    internal var TS60: Number? = null
    internal var TR05: Number? = null
    internal var TR25: Number? = null
    internal var TR60: Number? = null
    internal var RAIN_24H: Number? = null
    internal var RAIN_24: Number? = null
    internal var STID: String? = null
    internal var YEAR: Int? = null
    internal var MONTH: Int? = null
    internal var DAY: Int? = null
    internal var HOUR: Int? = null
    internal var MINUTE: Int? = null
    internal var SECOND: Int? = null


    override fun GetDefaultTempUnit(): UnitConverter.TempUnits {
        return UnitConverter.TempUnits.kCelsius
    }


    override fun GetDefaultLengthUnit(): UnitConverter.LengthUnits {
        return UnitConverter.LengthUnits.kMm
    }


    override fun GetDefaultSpeedUnit(): UnitConverter.SpeedUnits {
        return UnitConverter.SpeedUnits.kMps
    }


    override fun GetDefaultPressureUnit(): UnitConverter.PressureUnits {
        return UnitConverter.PressureUnits.kMb
    }
}